lisa/tests/lisa/test_events_checkers.py

```diff
@@ -0,0 +1,68 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2018, ARM Limited and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from unittest import TestCase
+from lisa.tests.lisa.utils import nullcontext
+
+from lisa.trace import TraceEventChecker, AndTraceEventChecker, OrTraceEventChecker, MissingTraceEventError
+
+""" A test suite for event checking infrastructure."""
+
+EVENTS_SET = {'foo', 'bar', 'baz'}
+
+class TestEventChecker(TestCase):
+    """
+    A test class that verifies checkers work as expected
+    """
+
+    def test_checkers(self):
+        and_1 = AndTraceEventChecker.from_events(['foo', 'bar'])
+        and_2 = AndTraceEventChecker.from_events(['foo', 'lancelot'])
+
+        or_1 = OrTraceEventChecker.from_events(['foo', 'bar'])
+        or_2 = OrTraceEventChecker.from_events(['foo', 'lancelot'])
+
+        # List of tuples: (<check expected to succeed>, <checker>)
+        checker_list = [
+            (True, TraceEventChecker('bar')),
+            (False, TraceEventChecker('non-existing-event')),
+
+            (True, and_1),
+            (False, and_2),
+
+            (True, or_1),
+            (True, or_2),
+            (False, OrTraceEventChecker.from_events(['arthur', 'lancelot'])),
+
+            (True, AndTraceEventChecker.from_events([and_1, or_1])),
+            (True, AndTraceEventChecker.from_events([and_1, or_2])),
+            (False, AndTraceEventChecker.from_events([and_1, and_2])),
+        ]
+
+
+        for expected_success, checker in checker_list:
```

Comment:
IIUC we're aggregating tests inside a single nosetest test. Shouldn't those above be split into their own tests methods?

Reply:
I split with one class per checker, so at least the x1.5 size increase brings the ability to easily test other methods than just `check_events()`